File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Systems/Mechaisms/Claw.java
package org.firstinspires.ftc.teamcode.Systems.Mechaisms;

import org.firstinspires.ftc.teamcode.Hardware.Constants.DepositConstants;
import org.firstinspires.ftc.teamcode.Hardware.Hardware;
import org.firstinspires.ftc.teamcode.Hardware.Wrappers.AnalogServo;

public class Claw {
    private AnalogServo servo;

    public Claw(Hardware hardware) {
        servo = new AnalogServo(hardware.claw, hardware.clawEnc);
    }

    public void setPosition(double position) {
        servo.setPos(position);
    }

    public double getPosition() {
        return servo.getPos();
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/koawalib/subsystems/Claw.kt
package org.firstinspires.ftc.teamcode.koawalib.subsystems

import com.acmerobotics.dashboard.config.Config
import com.asiankoala.koawalib.hardware.sensor.KDistanceSensor
import com.asiankoala.koawalib.hardware.servo.KServo
import com.asiankoala.koawalib.subsystem.Subsystem

class Claw(private val servo: KServo,
           private val distanceSensor: KDistanceSensor
) : Subsystem() {
    @Config
    companion object {
        @JvmField var closePos = 0.0
        @JvmField var openPos = 0.0
        @JvmField var distanceThreshold = 0.0
    }

    val readyToGrab get() = distanceSensor.lastRead < distanceThreshold

    fun setPos(pos: Double) {
        servo.position = pos
    }
}